### PR TITLE
ICU-21957 integrate CLDR release-42-alpha3 to ICU main for 72

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21957
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

Integrate CLDR release-42-alpha3 to ICU main for 72. Some notes:
- Emoji collation updates per CLDR-15965
- Unicode 15 collation data updates per CLDR-15516
- Arabic currency formats for 'latn' digits have been further updated so currency symbol follows number, per CLDR-15992
- Several updates to personNames data per CLDR-15977, including removing some spaces in formats for ja/lo/yue, and other changes to formats for zh/zh_Hant
- Compact decimal data for Japanese has been extended to values above 100000000000000 (i.e. above COMPACT_MAX_DIGITS=15) per CLDR-15966. There is nothing in the CLDR spec that disallows this; however ICU code cannot currently handle these new large values, so I have changed to code to skip loading them.
- Some formerly-provisional data has been promoted to contributed in certain locales, so it gets picked up by ICU.

### CODE REVIEW

**Review by markusicu — APPROVED** [review]
rslgtm

File: icu4j/main/classes/core/src/com/ibm/icu/impl/number/CompactData.java [hunk 1]
```diff
@@ -185,8 +185,10 @@ public void put(UResource.Key key, UResource.Value value, boolean isRoot) {
                 // Assumes that the keys are always of the form "10000" where the magnitude is the
                 // length of the key minus one. We expect magnitudes to be less than MAX_DIGITS.
                 byte magnitude = (byte) (key.length() - 1);
+                if (magnitude >= COMPACT_MAX_DIGITS) {
```
Comment (sffc) [inline icu4j/main/classes/core/src/com/ibm/icu/impl/number/CompactData.java:188]: It looks like this should hopefully "just work". There is a `largestMagnitude` variable, set on line 227 to the largest magnitude present in data. So if you increase `COMPACT_MAX_DIGITS` and other locales don't have the data, they should have a smaller `largestMagnitude` computed at runtime.

Comment (pedberg-icu) [reply icu4j/main/classes/core/src/com/ibm/icu/impl/number/CompactData.java:188]: I filed a separate ticket to make this change: https://unicode-org.atlassian.net/browse/ICU-22143

**Pull request merged**